packages/app/src/navigation/SidebarNavigationRow.vue

```diff
@@ -82,7 +82,24 @@ const props = withDefaults(defineProps <{
 })
 
 const badgeVariant = computed(() => {
-  return props.isNavBarExpanded ? 'ml-16px h-20px text-sm leading-3' : 'absolute outline-gray-1000 outline-2px outline bottom-0 left-36px text-xs h-16px leading-2'
+  const classes: string[] = []
+
+  if (props.isNavBarExpanded) {
+    classes.push('ml-16px', 'h-20px', 'text-sm', 'leading-3')
+  } else {
+    classes.push('absolute', 'outline-gray-1000', 'outline-2px', 'outline', 'bottom-0', 'text-xs', 'h-16px', 'leading-2')
+
+    // Right-align failure counts within sidebar (#25662)
+    if (props.badge.status === 'failed' || props.badge.status === 'error') {
+      // Add min-width so that single-digit badge overlaps sidebar icon at least a little
+      classes.push('right-4px', 'min-w-20px', 'text-center')
+    } else {
+      // Non-failures should left-align and overflow sidebar if needed
+      classes.push('left-36px')
+    }
+  }
```

Comment:
I interpreted Emil's comment in Slack to mean that the right alignment should only apply when there are three characters to display for the error count in the badge in the collapsed state (i.e. it is showing "99+"). The scenarios where it only shows 1 or 2 digits can stay in their original positions and not need the min width and would not extend past the collapsed sidebar's right border.

For this update, I think you could get away with only right aligning the badge if the value is `99+`, but you could also be a bit more generic and check the status as well as the length of the value that will show in the badge.

Reply:
Hmm, I see what you mean - I glossed over that last bullet point. This disagrees with the [original Figma](https://www.figma.com/file/PAOozuCxtfetqP8fUVvg0y/branch/QkCp2F69X0qyLo7qnYQ67e/Cypress-App?node-id=23899%3A229781&t=AeAPEJnkNCEES56e-1) for this which showed right-alignment for an `80` and `99+` case (but no single-digit case, hence this confusion).

Reply:
Updated in 8fb6cb92dc11c073ae3a249e29f4ce9b79f20425